Automattic/Gravatar-SDK-Android
# QuickEditor: Implement vertical content layout for the Avatar picker

Closes #270 

### Description

This PR implements the vertically scrolling list of Avatars in the picker. This is configurable via `GravatarQuickEditorParams`.

Two three required some modifications to make this work:
- The vertically scrolling section had to be done separately to include the title and the description in the Lazy grid with max span item count (to take full width). This makes them both part of the scrolling items and thus they go away on scroll down leaving more space for the avatars.
- `skipPartiallyExpanded` state of the bottom sheet. This is set to false for vertically scrolling and additionally for `compact` height devices (mainly phones in landscape) to match the iOS behavior.
- `MediaPickerPopup` had to be modified to allow for global positioning on the screen. Up until now, it could only be positioned within the parent composable. This is not enough with the Upload button being part of the scrollable content, because it can place on the edge of the parent layout leaving no space for the popup. 


https://github.com/user-attachments/assets/a822f371-4e92-41d1-8617-d7390c382458



### Testing Steps
**Important Note**: The backend is currently returning the wrong format for the `updated date` field. As a workaround, you can apply this patch to comment that field from the Avatar object.
[Gravatar-Android-13-04-48.patch](https://github.com/user-attachments/files/16455284/Gravatar-Android-13-04-48.patch)

2. Run the DemoApp 
3. Go to Avatar Update tab
4. Tap the `Vertical` option
5. Tap `Update Avatar` button
6. Follow the steps to log in (if necessary)
7. Play with the picker  - make sure the `+` button works properly and the popup shows at the right place.
8. Mock the `/avatars` request to return empty list of avatars (or just delete your avatars)
9. Open the QE again and confirm the empty state matches the `horizontal` layout. 
10. Type `upload image` button and confirm the popup is shown properly. 
11. Do some regression tests for the `horizontal` scroll as well 🙏
